### DISCUSSION TITLE
Getting evictions to error the build

### DISCUSSION THREAD

**Discussion by Rapierian** [sbt/sbt#7223] [Q&A]
Hello,

I've got these settings in my build:

`ThisBuild / evictionErrorLevel := Level.Error
ThisBuild / evictionWarningOptions := EvictionWarningOptions.full
ThisBuild / assumedEvictionErrorLevel := Level.Error`

And I've also tried `ThisProject` as well. Either way, when I run `sbt clean compile`, everything succeeds. But `sbt evicted` still shows plenty of evictions taking place.

**Comment by eed3si9n** [comment]
See https://www.scala-lang.org/blog/2021/02/16/preventing-version-conflicts-with-versionscheme.html

tl;dr is since sbt 1.5.0 I stopped using heuristics, and eviction error depends on opt-in information provided by versionScheme key by the library authors.

**Reply by Rapierian** [reply]
Hmm. So the reason I'm trying to track down evictions isn't so much to do with binary compatibility, but to do with cleaning up the information that shows up for security scanners - some of which are tripping on evicted libraries. I've almost got all of our evictions cleaned up, and I'd like to put in an alert to fail the build if any new evictions show up.

**Reply by Rapierian** [reply]
Which leads me back to my question. Is there a setting I can set to error the build on evictions?
